### PR TITLE
Use PyPI v1.4.0-1.4.3 to build windows for debugging purposes - detailed logs provided

### PR DESCRIPTION
**Pull request by bobleesj**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml
Comment (bobleesj) [inline recipe/meta.yaml:1]: The goal of this PR is to reproduce the following supporting Python 3.10-3.12 including win, linux, macOS

https://github.com/regro-cf-autotick-bot/diffpy.pdffit2-feedstock/blob/08ed7764c530bce6df949a2d07d488fbcfdd7fcd/recipe/meta.yaml

Screenshot 2024-09-26 at 8 47 50 AM

https://github.com/conda-forge/diffpy.pdffit2-feedstock/pull/13

Then, take it from here.

**Comment by bobleesj** [comment]
@conda-forge-admin, please rerender

**Comment by bobleesj** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (bobleesj) [reply recipe/meta.yaml:1]: Some action steps afterwards:

1. Fix windows build (current)
2. support Arm64 macOS
3. release for the community (only if it is urgent - discuss with SB)
4. Implement CI (windows, linux, ubuntu, etc) for the version that works in 1.
5. re-cookiecutter from the version

Comment (bobleesj) [reply recipe/meta.yaml:1]: While waiting for CI...

1. use windows local machine to build 1.4.0 PyPI `sdist` (note how to reproduce)
2. use macOS intel ...
3. use macOS M1 and M2 ...

**Comment by bobleesj** [comment]
Reference: https://github.com/diffpy/diffpy.pdffit2/releases

Background info:
- No PyPI v1.4.2 provided
- v1.4.1 fixes C++/path issues [CHANGELOG](https://github.com/diffpy/diffpy.pdffit2/blob/1.4.1/CHANGELOG.md)
- v1.4.3 supports Python 3.11-12 with no "functional change from 1.4.1" [CHANGELOG](https://github.com/diffpy/diffpy.pdffit2/blob/1.4.3/CHANGELOG.md)
- v1.4.4 only has rc on PyPI - cookiecutting not finished

**240926 8:40 AM  rollback report 1**
use v1.4.0 b8 to reproduce earlier build success:

- good: wins, macOS (intel), linux for Python 3.9-12 build
- bad: still not using PyPI v1.4.1, 1.4.3

**240926 9:11 AM  rollback report 2**
use PyPI v1.4.1 https://github.com/conda-forge/diffpy.pdffit2-feedstock/pull/15 

- good wins, macOS (intel), linux for Python 3.9-12 build
- bad: macOS arm64

**240926 9:26 AM  rollback report 3**
use PyPI v1.4.3 , keep bumping the version https://github.com/conda-forge/diffpy.pdffit2-feedstock/pull/26/commits/27bd6524082c342f53f8a85baca4826bc6c7ad86

- skip build for python < 3.10 
- remove `preserve_egg_dir: true` and `gsl` in `host:` and `run:` https://github.com/conda-forge/diffpy.pdffit2-feedstock/pull/26/commits/7db7dd13b53e350c4a0284664581facd6db18108

🔴 **IMPORTANT:** 🔴 Error detected with the latest change.

```
(base) %SRC_DIR%>conda activate "%PREFIX%" 
    from diffpy.pdffit2 import pdffit2
ImportError: DLL load failed while importing pdffit2: The specified module could not be found.
```
- good: identified which changes caused windows fail 1) version 2) egg_dir 3), or 3) gsl
- bad: windows failing

**240926 10:00 AM  rollback report 4**
use PyPI v1.4.3 , do NOT remove `preserve_egg_dir: true`, only change bump v1.4.1 to v1.4.3 while remove `gsl` in `host:` and `run:`

- bad: still wins fails
- good: we can further narrow down the error source either 1) version or 2) gsl

**240926 10:21 AM  rollback report 5**
use PyPI v1.4.3, no change from report 2, expect to pass..

- good: 🟢 PyPI v1.4.3 builds for windows below with the setup below.
```
  host:
    - python
    - pip
    - setuptools
    - gsl {{ gsl }}
  run:
    - python
    - setuptools
    - six
    - diffpy.structure
    - gsl {{ gsl }}
```

It is odd that `gsl` must be under `host:` and `run:` for windows, otherwise we get `ImportError: DLL load failed while importing pdffit2:`. This is not the current main priority. Let's do a final round of this investigation of windows building by removing `egg_dir_true`. Good progress is made. 

- bad: do we still need `preserve_egg_dir`? 

**230926 10:21 AM  rollback report 6**
Use same setup as report 5, just remove `preserve_egg_dir`

- good: 🟢 we do not need `preserve_egg_dir=true` for v1.4.3. to build, that means it's the lack of {{ gsl }} that caused the `DLL load` issue in report 4.
Screenshot 2024-09-26 at 10 55 51 AM

- bad: macOS arm64 fails but this is the next priority

Let's keep the latest commit: https://github.com/conda-forge/diffpy.pdffit2-feedstock/pull/26/commits/e1a41fc4365d557b7710f7c91b6ad7856c7f7730

**Comment by bobleesj** [comment]
@conda-forge-admin, please rerender

**Comment by bobleesj** [comment]
@conda-forge-admin, please rerender

**Comment by bobleesj** [comment]
@conda-forge-admin, please rerender

**Comment by bobleesj** [comment]
@conda-forge-admin, please rerender

**Comment by bobleesj** [comment]
Closing this PR since we know the exact setup required to build windows.

**Comment by sbillinge** [comment]
wow, amazing work!  Thanks @bobleesj

**Pull request closed without merging**